src/acceleration_ros_publisher.cpp

```diff
@@ -58,22 +58,39 @@ void AccelerationRosPublisher::run()
         //std::cout << "thread " << this_id << " running...\n";
         //RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
 
-        auto started = std::chrono::high_resolution_clock::now();
-        bool success = false;
-        m_message = m_dataProvider->getData(success);
-        auto done = std::chrono::high_resolution_clock::now();
+        int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
 
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Publishing acceleration x y z in time %d ms: '%f' '%f' '%f'",
-                    m_message.linear_acceleration.x, m_message.linear_acceleration.y, m_message.linear_acceleration.z ,
-                    std::chrono::duration_cast<std::chrono::milliseconds>(done-started).count());
+        switch(operation_mode) {
+        case DEVICE_CONTINUOUS_SAMPLING_MODE:
+        {
+            m_dataProvider->load();
+            auto started = std::chrono::high_resolution_clock::now();
+            bool success = false;
+            m_message = m_dataProvider->getData(success);
+            auto done = std::chrono::high_resolution_clock::now();
 
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Publishing acceleration x y z in time %d ms: '%f' '%f' '%f'",
+                        m_message.linear_acceleration.x, m_message.linear_acceleration.y, m_message.linear_acceleration.z ,
+                        std::chrono::duration_cast<std::chrono::milliseconds>(done-started).count());
+
+
+            if(success)
+                m_publisher->publish(m_message);
+            count++;
```

Comment:
why is this increased? is it used anywhere? if not, please remove it

Reply:
Coding style will be applied in a clean commit